OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -704,6 +704,18 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Moved ITargetablePositions from Building to HitShape
+				if (engineVersion < 20170523)
+				{
+					var building = node.Value.Nodes.FirstOrDefault(n => n.Key == "Building");
+					if (building != null)
+					{
+						Console.WriteLine("The 'Building' trait no longer provides per-cell targetable positions.");
```

Comment:
IMO it would be better to unconditionally add a `HitShape` node with `UseFootprintOffsets: true`.  It will then merge in with any other `HitShapes` and hopefully just work.